### PR TITLE
Added the wboss timer module

### PR DESCRIPTION
**Pull request by kallith**
Base functionality implemented, still missing future death checks and reloading times from file.

Additionally added a method to the handler that is called after the client is ready.

### CODE REVIEW

**Comment by kallith** [comment]
Oh great my editor changed file endings

File: handlers/world_boss_timers.py [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        d = datetime(dt.year, 11, 1)
+        # ends before the last sunday of october
+        self.dstoff = d - timedelta(days=d.weekday() + 1)
+        if self.dston <= dt.replace(tzinfo=None) < self.dstoff:
+            return timedelta(hours=1)
+        else:
+            return timedelta(0)
+    def tzname(self, dt):
+        return "GMT +1"
+
```
Comment (RoarkGit) [inline handlers/world_boss_timers.py]: classes should be separated by 2 newlines

File: handlers/world_boss_timers.py [hunk 2]
```diff
@@ -0,0 +1,127 @@
+from collections import defaultdict
+from datetime import datetime, timedelta, tzinfo, timezone
+import re
+
+from .handler import Handler
+
```
Comment (RoarkGit) [inline handlers/world_boss_timers.py]: + one newline

File: handlers/world_boss_timers.py [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+class GMT1(tzinfo): # directly from pydocs B-)
+    def utcoffset(self, dt):
+        return timedelta(hours=1) + self.dst(dt)
```
Comment (RoarkGit) [inline handlers/world_boss_timers.py]: Separate class-level functions by 1 newline

File: handlers/world_boss_timers.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+                dt = self.timestr_to_datetime(timestr)
+            except Exception as e:
+                return 'Parsing error: could not interpret "{}"'.format(timestr)
+            self._wb_tod[key] = dt
+            return 'Changed TOD of {} to {:%Y-%m-%d %H:%M}'.format(key, dt)
+        else:
+            return 'Parsing error: key {} does not match any of {}'.format(key,
+                                                ", ".join(self._wb_tod.keys()))
+
+    def timestr_to_datetime(self,timestr):
```
Comment (RoarkGit) [inline handlers/world_boss_timers.py]: Space between self, timesrt

File: handlers/world_boss_timers.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+                d,h = diff.days, diff.seconds//3600+1
+                additional = "closes in {}d {}h".format(d,h)
+            else:
+                additional = "window has passed"
+            timer_lines.append(line_fmt.format(key,start_dt,end_dt,additional))
+        return response.format(servertime=cur_dt, lines="\n".join(timer_lines))
+
+    def command_triggers(self):
+        return ['!tod', '!timers']
+
```
Comment (RoarkGit) [inline handlers/world_boss_timers.py]: Remove main before submit

File: nsabot.py [hunk 6]
```diff
@@ -12,16 +12,19 @@ class NSABot(Client):
...
             handler.register_bot(self)
 
     async def on_ready(self):
         print('Connected.')
+        for handler in self.handlers:
+            handler.client_ready()
 
     async def on_message(self, message):
-        if message.content:
+        if message.content and message.author != self.user:
```
Comment (RoarkGit) [inline nsabot.py]: What is the point of this line change? Does it ever trigger its own commands?

Comment (kallith) [reply nsabot.py]: It can if it would ever send a message, with one of the triggers.

File: handlers/world_boss_timers.py (hunk 5 above)
Comment (kallith) [reply handlers/world_boss_timers.py]: Erhm, whoops.

File: handlers/world_boss_timers.py (hunk 2 above)
Comment (kallith) [reply handlers/world_boss_timers.py]: I don't get it, shouldn't stdlib imports be clumped together?

Comment (RoarkGit) [reply handlers/world_boss_timers.py]: It's kind of annoying but there should be 2 lines (so one additional line) after imports and before top-level definitions.

Comment (kallith) [reply handlers/world_boss_timers.py]: Alright I see, should be correct now if I understood you correctly

**Pull request merged**